.github/workflows/build.yml

```diff
@@ -61,13 +61,23 @@ jobs:
         java-version: 1.8
         server-id: github # Value of the distributionManagement/repository/id field of the pom.xml
         settings-path: ${{ github.workspace }} # location for the settings.xml file
+    - name: Detect the version of Hibernate Reactive
+      id: detect-version
+      run: |
+        ./gradlew hibernate-reactive-core:properties | grep '^version:' | sed -E 's/version: (.*)/::set-output name=version::\1/'
+    - name: Generate a unique version
+      id: generate-unique-version
+      env:
+        UNIQUE_VERSION: ${{ format( '{0}-{1}', steps.detect-version.outputs.version, github.sha ) }}
+      run: echo "::set-output name=version::$UNIQUE_VERSION"
     - name: Create maven package
-      run: ./gradlew assemble publishToMavenLocal 
-
-    # The USERNAME and TOKEN need to correspond to the credentials environment variables used in
-    # the publishing section of your build.gradle
-    #- name: Publish to GitHub Packages
-    #  run: gradle publish
-    #  env:
-    #    USERNAME: ${{ github.actor }}
-    #    TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      env:
+        ORG_GRADLE_PROJECT_projectVersion: ${{ steps.generate-unique-version.outputs.version }}
+      run: ./gradlew assemble publishToMavenLocal
+    - name: Publish to GitHub Packages (SNAPSHOT only, push only, PR excluded)
+      env:
+        ORG_GRADLE_PROJECT_projectVersion: ${{ steps.generate-unique-version.outputs.version }}
+        ORG_GRADLE_PROJECT_githubUser: ${{ github.actor }}
+        ORG_GRADLE_PROJECT_githubToken: ${{ github.token }}
+      if: endsWith( steps.detect-version.outputs.version, '-SNAPSHOT' ) && github.event_name == 'push'
+      run: ./gradlew publish -x test
```

Comment:
If tests are missing, doesn't this mean that there are going to be test failures?

Reply:
It's not a release, just a snapshot.

This runs after we run `./gadlew publishToMavenLocal` (in one of the previous steps), so unit tests (if any) have already run. I'll just remove this `-x test`, it's not useful.

Regarding integration tests... I suppose I could move this to a separate job and only publish after all other jobs have run.
That being said, I'm not sure it matters much: do you expect code merged to master to fail tests very often? Worst case, we just publish a snapshot that doesn't work, which means we'll publish another one soon after.

Reply:
I was just trying to understand the flow. I don't think that's intuitive when looking at the file.

Reply:
To be fair, enabling the tests won't test on all the dbs anyway.